djcelery_email/utils.py

```diff
@@ -73,14 +73,21 @@ def email_to_dict(message):
 
 
 def dict_to_email(messagedict):
-    messagedict = copy.deepcopy(messagedict)
-    extra_attrs = {}
+    message_kwargs = copy.deepcopy(messagedict)  # prevents missing items on retry
+
+    # remove items from message_kwargs until only valid EmailMessage/EmailMultiAlternatives kwargs are left
+    # and save the removed items to be used as EmailMessage/EmailMultiAlternatives attributes later
+    message_attributes = ['content_subtype', 'mixed_subtype']
     if settings.CELERY_EMAIL_MESSAGE_EXTRA_ATTRIBUTES:
-        for attr in settings.CELERY_EMAIL_MESSAGE_EXTRA_ATTRIBUTES:
-            if attr in messagedict:
-                extra_attrs[attr] = messagedict.pop(attr)
-    attachments = messagedict.pop('attachments')
-    messagedict['attachments'] = []
+        message_attributes.extend(settings.CELERY_EMAIL_MESSAGE_EXTRA_ATTRIBUTES)
+    attributes_to_copy = {}
```

Comment:
Could simplify further as:

```suggestion
    attributes_to_copy = {att: message_kwargs.pop(att) for att in message_attributes if att in message_kwargs}
```

Probably less readable though. I'm fine either way.

Reply:
Agreed on it being less readable. I'm not sure the performance benefits of the dict comprehension would matter much since this loop won't usually be going through very many items.